crates/oxc_linter/src/rules/eslint/max_len.rs

```diff
@@ -0,0 +1,134 @@
+use serde_json::Value;
+
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(max-len): Line is too long (current length: {current_length:?}, maximum: {max:?})")]
+#[diagnostic(
+    severity(warning),
+    help("Consider breaking this line into multiple lines or shortening comments/codes where applicable")
+)]
+struct MaxLenDiagnostic {
+    current_length: usize,
+    max: usize,
+    #[label]
+    span: Span,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct MaxLen(Box<MaxLenConfig>);
+
+#[derive(Debug, Clone)]
+pub struct MaxLenConfig {
+    max: usize,
+}
+
+impl std::ops::Deref for MaxLen {
+    type Target = MaxLenConfig;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Default for MaxLenConfig {
+    fn default() -> Self {
+        Self { max: 80 } // default max length is often considered to be 80 characters
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforce a maximum length of characters per line
+    ///
+    /// ### Why is this bad?
+    /// Lines that are too long can be difficult to read, understand, and maintain.
+    /// Excessively long lines can also result in horizontal scrolling which hinders readability.
+    /// While there is no objective maximum length considered acceptable for a line,
+    /// a commonly used standard is 80 characters per line.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // This line is fine
+    /// const example = "This is a concise example.";
+    ///
+    /// // This line might be considered too long and difficult to read without wrapping or horizontal scrolling
+    /// const tooLongExample = "This line is an example of a line that stretches far beyond the conventional length and could be hard to read.";
+    /// ```
+    MaxLen,
+    pedantic
+);
+
+impl Rule for MaxLen {
+    fn from_configuration(value: Value) -> Self {
+        let max = value
+            .as_number()
+            .and_then(serde_json::Number::as_u64)
+            .map_or(80, |v| usize::try_from(v).unwrap_or(80)); // default max length of 80 chars if not specified
+
```

Comment:
Looks like this rule has more [options](https://eslint.org/docs/latest/rules/max-len#options).

Reply:
> Looks like this rule has more [options](https://eslint.org/docs/latest/rules/max-len#options).

ok